code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/xdm/XDMMediaSchema.kt

```diff
@@ -0,0 +1,34 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.media.internal.xdm
+
+import com.adobe.marketing.mobile.util.TimeUtils
+import java.util.Date
+
+internal data class XDMMediaSchema(
+    var eventType: XDMMediaEventType,
+    var timestamp: Date,
+    var mediaCollection: XDMMediaCollection
+) : XDMProperty {
+
+    override fun serializeToXDM(): Map<String, Any> {
+        val map = mutableMapOf<String, Any>()
+
+        map["eventType"] = XDMMediaEventType.getTypeString(eventType)
```

Comment:
nit: should we add these keys “eventType” “timestamp” “mediaCollection” as Constants?